src/MySqlConnector/Core/SchemaProvider.cs

```diff
@@ -987,7 +1032,34 @@ private async Task FillDataTableAsync(IOBehavior ioBehavior, DataTable dataTable
 		using (var command = m_connection.CreateCommand())
 		{
 #pragma warning disable CA2100
-			command.CommandText = "SELECT " + string.Join(", ", dataTable.Columns.Cast<DataColumn>().Select(static x => x!.ColumnName)) + " FROM INFORMATION_SCHEMA." + tableName + ";";
+			command.CommandText = "SELECT "
+				+ string.Join(", ", dataTable.Columns.Cast<DataColumn>().Select(static x => x!.ColumnName))
+				+ " FROM INFORMATION_SCHEMA." + tableName;
+
+			if (restrictionValues.Any())
+			{
+				StringBuilder sbWhere = new StringBuilder();
+				List<string> whereParts = new List<string>(4);
+
+				if (restrictionValues.Length >= 1 && !string.IsNullOrEmpty(restrictionValues[0]))
+					whereParts.Add($"UPPER(TABLE_CATALOG) = '{restrictionValues[0]?.ToUpperInvariant()}'");
```

Comment:
* This doesn't handle a literal `'` in the restriction value (and will generate a SQL parsing error or, worse, a SQL injection bug).
  * It's possible this is bug-compatible with MySql.Data (I haven't checked), but we shouldn't allow security vulnerabilities in order to do so
* Why `UPPER`? This will fail for case-sensitive identifiers, i.e., `lower_case_table_names = 0`.
* The first restriction is not always `TABLE_CATALOG`; the set of restrictions varies according to the schema being retrieved. For example: for `"Users"`, `restrictions[0]` is a pattern to match against the user name; for `"Procedures"`, `restrictions[1]` is `ROUTINE_SCHEMA`.
  * This appears to be very poorly documented for MySql.Data, but I'm wary of being gratuitously different here; at a minimum, we should provide a porting guide mapping MySql.Data restriction indexes onto MySqlConnector restriction indexes (if there's a good reason to be different).
* `restrictionValues[0]?` is unnecessary if it's already been tested with `string.IsNullOrEmpty`.

Reply:
- Literal `'` : correct, but i think this isn't the correct layer for injection-prevention
- Upper: case-sensitive database configuration and duplicated-name-using are insane. If we remove upper then we are always case-sensitive.
- The restrictions-mapping is a kind of magic. I implemented it only for some known "`GetSchema()`"-calls. However, these are already sufficient for using NHibernate's schema generator. I will comment this in the code.
- restrictionValues[0]? -> "`string?`" can be `STRING.EMPTY` or not?

Reply:
here is a docu https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/schema-restrictions of the restrictions-array

Reply:
- lower_case_table_names: In the future, we should read the server configurations such as lower_case_table_names or max_allowed_packet in order to dynamically adjust the connector. See https://github.com/mysql-net/MySqlConnector/blob/683ccf2c932255cab15e98b0522f9818e55a435c/src/MySqlConnector/Core/CommandExecutor.cs#L65